### PR TITLE
Adding clear btn callback

### PR DESCRIPTION
**Pull request by anonymous-ME**
**What issue does this pull request resolve?**
onInputChange not called when input cleared. Fix for this issue [#594](https://github.com/ericgio/react-bootstrap-typeahead/issues/594)

**What changes did you make?**
added the callBack

**Is there anything that requires more attention while reviewing?**
nope

### CODE REVIEW

File: src/core/Typeahead.js [hunk 1]
```diff
@@ -261,6 +261,9 @@ export function getInitialState(props: Props): TypeaheadState {
 }
 
 export function clearTypeahead(state: TypeaheadState, props: Props) {
+  if (typeof props.onInputChange === "function") {
+    props.onInputChange("");
```
Comment (ericgio) [inline src/core/Typeahead.js:265]: As noted, `onInputChange` passes the change event as the second argument. It also doesn't seem like a good idea to trigger a callback during a state change. That should happen after the state change is complete.

**Review by ericgio — CHANGES_REQUESTED** [review]
Hey @anonymous-ME, thanks for submitting this PR. The issue you're trying to solve with this change is actually a bit more complicated than it first appears. `onInputChange` also passes the change event as the second argument, but that event doesn't actually happen, since the value changes in response to a click on a different element.

**Comment by ericgio** [comment]
Oh, and if you'd like to continue pursuing this change, some tests would be great :)

**Comment by anonymous-ME** [comment]
> Hey @anonymous-ME, thanks for submitting this PR. The issue you're trying to solve with this change is actually a bit more complicated than it first appears. `onInputChange` also passes the change event as the second argument, but that event doesn't actually happen, since the value changes in response to a click on a different element.

Hi @ericgio, what solution would you suggest in this case?

**Comment by anonymous-ME** [comment]
Should we add a callback for onClear instead?

**Comment by ericgio** [comment]
I think the solution is that the event needs to be manually triggered by the clear callback, which would then trigger `onInputChange` as one would expect.

**Comment by anonymous-ME** [comment]
Hi @ericgio, I'm thinking of doing something like this to simulate onChange for TypeaheadManager. 

```
  const event = new Event('change');
  element.dispatchEvent(event);
```

But not sure how to get TypeaheadManager element reference inside of clearTypeahead callback.

**Comment by ericgio** [comment]
This was added in v5.2

**Pull request closed without merging**